### PR TITLE
 Container signing

### PR DESCRIPTION
**Pull request by MilanPospisil**
Issue: AAH-1265

https://issues.redhat.com/browse/AAH-1265

Depends on https://github.com/ansible/galaxy_ng/pull/1414, https://github.com/ansible/galaxy_ng/pull/1434

Docs:
https://hackmd.io/@rochacbruno/SkQmp8-09

### CODE REVIEW

**Comment by rochacbruno** [comment]
@MilanPospisil notice that there is a case when container is not a `push` but `synced` from a remote and those containers needs an extra parameter on the sign/ api as explained on https://hackmd.io/@rochacbruno/SkQmp8-09#1-Images-synced-from-remotes

File: src/api/sign-containers.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      postObj['future_base_path'] = 'library/busybox';
+    }
+
+    return this.http.post(
+      `/repositories/container/${pulp_type}/${containerId}/sign/`,
+      postObj,
+    );
+  }
+
+  /*public getRepository(repoName: string) {
```
Comment (himdel) [inline src/api/sign-containers.ts:20]: commented out code

File: src/utilities/repo-signing.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    if (strings.length >= 1) {
+      pulp_id = strings[strings.length - 1];
+
+      if (strings.length >= 2 && strings[strings.length - 1] == '') {
+        pulp_id = strings[strings.length - 2];
+      }
+    }
+
+    return pulp_id;
+  }
```
Comment (himdel) [inline src/utilities/repo-signing.ts:21]: `parsePulpIDFromURL` from `src/utilities`

File: src/utilities/repo-signing.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    SignContainersAPI.getSigningService(service)
+      .then((result) => {
+        const pulp_href = result.data.results[0]['pulp_href'];
+        return SignContainersAPI.sign(
+          item.pulp.repository.pulp_id,
+          RepoSigningUtils.getContainerPulpType(item),
+          pulp_href,
+        ).then((result) => {
+          addAlert({
+            id: 'loading-signing',
```
Comment (himdel) [inline src/utilities/repo-signing.ts:69]: probably want to remove the id unless we need to find this alert later in code

**Comment by himdel** [comment]
Test failure unrelated, fixing in https://github.com/ansible/ansible-hub-ui/pull/2503

File: src/utilities/repo-signing.ts (hunk 2 above)
Comment (MilanPospisil) [reply src/utilities/repo-signing.ts:21]: Yes, Im using this method, but I forgot to delete the unused code (strange that compiler did not noticed it).

**Comment by himdel** [comment]
Test failures in `container-signing.js` are happening because nobody enabled container signing on CI.
Looks like https://galaxyng.netlify.app/config/container_signing/ documents how to enable it, if anyone can add this, please do, if not, I'll look into it after PTO.

**Review by himdel — APPROVED** [review]
LGTM :)

(but don't merge until CI issue is resolved)

**Comment by MilanPospisil** [comment]
https://github.com/ansible/galaxy_ng/pull/1434 must be merged first in order to merge this.

**Review by rochacbruno — APPROVED** [review]

File: src/api/sign-containers.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class API extends PulpAPI {
+  public getSigningService(serviceName: string) {
+    return this.http.get(`/signing-services/?name=${serviceName}`);
+  }
+
+  public sign(containerId: string, pulp_type, signServicePath) {
+    const postObj = { manifest_signing_service: signServicePath };
+    if (pulp_type == 'container') {
+      postObj['future_base_path'] = 'library/busybox';
```
Comment (himdel) [inline src/api/sign-containers.ts:11]: this should probably not be hardcoded to library/busybox here

**Pull request merged**